### ISSUE TITLE
Replace unwrap calls with Results

### ISSUE THREAD

**Issue by rgardner** [rustyhorde/readline-sys#6]
| unwrap | expect 
------------|-----------|-----------
lib.rs       |   8          |    1 
build.rs   |  5           |   0     

It is not advised to use `unwrap` or `expect` in library code, unless it is trivially true that it won't panic. Instead, these should be swapped out for `Result`s that describe the nature of the issue. These are breaking changes in the interface to the crate, but I think other library authors and application developers will appreciate the added safety.

**Comment by CraZySacX** [comment]
I agree.  I'll make a pass at better safety.

**Comment by rgardner** [comment]
Awesome! Thanks @CraZySacX

**Comment by CraZySacX** [comment]
Fixed with fbeac77ff410f5cb790b39e31f39f4aba8635c5e.  Let me know if you find any issues.

**Comment by rgardner** [comment]
Thanks Jason! The one API change I disagree with is that `readline` returns
an `Err` when a user enters `ctrl-d`. I think the return type would be
improved by making it `Result<Option<String>, ...>`, because it's not
necessarily an error for the user to enter `ctrl-d`, that is in fact
expected behavior in a lot of scenarios. I think `Result<Option>` covers
the semantics better. Up to you though.

**Comment by CraZySacX** [comment]
Yeah I tossed that one around myself.  I originally implemented it that way, but I'm wasn't sure I liked forcing the client implementation to do 2 checks.

```rust
    let prompt = String::from("$ ");
    loop {
        let response = match rl_sys::readline(prompt) {
            Ok(o)  => match o {
                Some(s) => s,
                None    => break,
            },
            Err(_) => break,
        };
        println!("{}", response);
    }
```

After thinking about it some more, I think you are right.  ```Result<Option<String>, ...>``` better conveys what is actually happening.

**Comment by rgardner** [comment]
I totally understand the hesitation. The client doesn't have to do two
levels of nesting though (which makes things a little better). This is what
I would replace the README example with:

```rust
let prompt = String::from("$ ");

loop {
    let response = match rl_sys::readline(prompt.clone()) {
        Ok(Some(s)) => s,
        Ok(None) => break,
        Err(e) => panic!("encountered an error: {:?}", e),
    };
    println!("{}", response);
}
```

Note that I added `prompt.clone()`. This code didn't actually compile
before because prompt was moved into `readline`. This fixes that issue.
Alternatives are that prompt could be declared in the loop or the readline
signature could take an `&str` instead of a String.
